sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/SpecificParquetRecordReaderBase.java

```diff
@@ -93,48 +93,17 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptCont
       throws IOException, InterruptedException {
     Configuration configuration = taskAttemptContext.getConfiguration();
     ParquetInputSplit split = (ParquetInputSplit)inputSplit;
+    Preconditions.checkState(split.getRowGroupOffsets() == null,
```

Comment:
Why don't you just simply use `assert`?

Reply:
The assert of Java code depends on `-ea` flag to activate,  we can't guarantee users execute spark apps with this flag, or do we just want to ensure this `assert` works in UTs?

Reply:
Either is reasonable here. If it should never be true in any execution of Spark, we assert it. If user input or state can cause it to not be true, we need something besides an assert. Most of the point is verifying this at test time, yes.

Reply:
For the current spark code, I think `split.getRowGroupOffsets()` should always be null

Reply:
https://github.com/apache/spark/blob/61881bb6988aa0320b4bacfabbc0ee6f05f287cb/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala#L255-L267

Reply:
https://github.com/apache/spark/blob/61881bb6988aa0320b4bacfabbc0ee6f05f287cb/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetPartitionReaderFactory.scala#L117-L134